tsdb/functions.go

```diff
@@ -1320,19 +1466,18 @@ func MapTopBottom(itr Iterator, c *influxql.Call) interface{} {
 
 	// buffer so we don't allocate every time through
 	var pp PositionPoint
-	if len(c.Args) > 2 {
+	if argCount > 2 {
```

Comment:
`args` should perhaps be renamed to `fields`, as it is just the fields being asked for.

For `TOP` you can specify `select top(value, 5)` or `select top(value, host, 5)` and it will change how it determines results.  So you need to know how many fields exist while examining data.  This is typically one less than the arg count which includes the `limit`

I'll change the variable to be more meaningful.  I didn't like it when I did it either.